docs/design/2018-07-22-enhance-propagations.md

```diff
@@ -0,0 +1,189 @@
+# Proposal: Enhance constraint propagation in TiDB logical plan
+
+- Author(s):     [@bb7133](https://github.com/bb7133), [@zz-jason](https://github.com/zz-jason) 
```

Comment:
This proposal was mainly completed by you, please remove me from the author list 😂

Reply:
Well, the proposal #4 is all from your issue :)